web/src/main/java/se/inera/intyg/intygstjanst/web/service/impl/CertificateExportServiceImpl.java

```diff
@@ -0,0 +1,130 @@
+        SAXException {
+        final var resources = resourceResolver.getResources(TEXTS_LOCATION);
+        final var textFiles = Arrays.stream(resources).filter(this::isTextFile).collect(Collectors.toList());
+        return getCertificateTexts(textFiles);
+    }
+
+    @Override
+    public CertificateExportPageDTO getCertificateExportPage(String careProviderId, int page, int size) {
+        final var pageable = PageRequest.of(page, size, Sort.by(Direction.ASC, "signedDate", "id"));
+        final var certificatePage = certificateRepository.findCertificatesForCareProvider(careProviderId, pageable);
+        final var certificates = certificatePage.getContent();
+        final var totalCertificates = certificatePage.getTotalElements();
+        final var totalRevoked = certificateRepository.findTotalRevokedForCareProvider(careProviderId);
+        final var certificateXmls = getCertificateXmls(certificates);
+        final var certificateCount = certificatePage.getNumberOfElements();
+        return new CertificateExportPageDTO(careProviderId, page, certificateCount, totalCertificates, totalRevoked,  certificateXmls);
+    }
+
+    private ArrayList<CertificateTextDTO> getCertificateTexts(List<Resource> textFiles) throws IOException, ParserConfigurationException,
+        SAXException, TransformerException {
+        final var certificateTexts = new ArrayList<CertificateTextDTO>();
+        for (final var textFile : textFiles) {
+            final var textDocument = parseTextFile(textFile);
+            final var type = getTextAttribute(textDocument, TYPE_ATTRIBUTE);
+            final var version = getTextAttribute(textDocument, VERSION_ATTRIBUTE);
+            final var xml = getTextXml(textDocument);
+            certificateTexts.add(new CertificateTextDTO(type, version, xml));
+        }
+
+        return certificateTexts;
+    }
+
+    private Document parseTextFile(Resource textFile) throws IOException, ParserConfigurationException, SAXException {
+        return DocumentBuilderFactory.newInstance().newDocumentBuilder().parse(textFile.getInputStream());
+    }
+
+    private String getTextAttribute(Document textDocument, String attribute) {
+        return textDocument.getDocumentElement().getAttribute(attribute);
+    }
+
+    private boolean isTextFile(Resource resource) {
+        return resource.getFilename() != null && resource.getFilename().endsWith(XML_FILE_EXTENSION);
+    }
+
+    private String getTextXml(Document textDocument) throws TransformerException {
+        final var stringWriter = new StringWriter();
+        final var transformer = TransformerFactory.newInstance().newTransformer();
+        transformer.transform(new DOMSource(textDocument), new StreamResult(stringWriter));
+        return stringWriter.toString();
+    }
+
+    private ArrayList<CertificateXmlDTO> getCertificateXmls(List<Certificate> certificates) {
```

Comment:
You could make use of the Stream api to do these kind of transformations.

private List<CertificateXmlDTO> getCertificateXmls(List<Certificate> certificates) {
        return certificates.stream()
            .map(certificate -> new CertificateXmlDTO(
                certificate.getId(),
                certificate.getCertificateMetaData().isRevoked(),
                certificate.getOriginalCertificate().getDocument()
            ))
            .collect(Collectors.toList());
    }